### DISCUSSION TITLE
Autostart in Windows 10

### DISCUSSION THREAD

**Discussion by AndyZaa** [itsToggle/plex_debrid#525] [Q&A] [answered]
Just installed plex_debrid on my Windows 10 Plex machine and it is brilliant! Now I'm trying to figure out autostart routine for both rclone and plex_debrid for when I reboot. The Discord link in the instructions doesn't connect and I can't find plex_debrid on Discord. Can anyone point me to how to autostart in Win 10 and/or where to find the Discord (or other) community? Thanks.

**Comment by JSS3M3** [comment]
Same issue.

**Comment by AndyZaa** [comment]
Installed Jackett today which looks awesome as well. One of the feeds needed FlareSolverr so installed that as well. Everything working thanks to your comprehensive instructions, @itsToggle! Thank you again for this stellar code.

My one open set of related questions is do I need to / how to autostart the various components after an inevitable reboot. That includes:
* plex_debrid
* rclone
* Jackett
* FlareSolverr

Cheers and thanks to all who might help!

**Comment by PittsGitHub** [comment, marked as answer]
Not sure what the official suggestion is and I may have made it work in a hacky way.I wrote two simple Powershell files that are called on start-up and set plex-debrid to open without the menu prompts.

This has led to when the PC is turned on and logged into plex-debird auto starts and plex auto starts anyway.

**plex debrid script:**
```
$Title = "plex-Debrid"

$host.UI.RawUI.WindowTitle = $Title

cd <YOUR FULL PLEX-DEBRID FOLDER PATH HERE>

python ./main.py
```

**RClone Script:**
```
$Title = "rclone"

$host.UI.RawUI.WindowTitle = $Title

cd <YOUR FULL RCLONE FOLDER PATH HERE>

./rclone.exe cmount realdebrid: X: --dir-cache-time 10s
```

**Reply by AndyZaa** [reply]
Thanks @PittsGitHub. I'm going to wait a day or two to see if there are any other techniques, then will employ your method.

**Comment by AndyZaa** [comment]
Thanks again @PittsGitHub. I used your approach with a minor edit replacing "cd" with "Set-Location".

plex debrid script:

$Title = "plex-debrid"
$host.UI.RawUI.WindowTitle = $Title
Set-Location E:\Desktop\plex_debrid-main
python ./main.py

RClone Script:

$Title = "rclone"
$host.UI.RawUI.WindowTitle = $Title
Set-Location E:\Desktop\rclone
./rclone.exe cmount realdebrid: X: --dir-cache-time 10s

Then I created a batch file and put it into task scheduler to run on computer start. I had to put the execution policy modifiers in to get it past the policy restrictions. Seems to be working fine.

Autostart-rclone-debrid.bat

powershell.exe -executionpolicy remotesigned -File E:\Desktop\Powershell_Scripts\rclone.ps1
powershell.exe -executionpolicy remotesigned -File E:\Desktop\Powershell_Scripts\plex-debrid.ps1

**Reply by AndyZaa** [reply]
Oh yeah, and I just noticed that a small typo crept into your plex debrid script in the first line:
`$Title should be $Title

**Reply by PittsGitHub** [reply]
Thank you, fixed :)

**Comment by AndyZaa** [comment]
Just wanted to revisit and document what I ended up employing for autostarting. I had some reliability problems with my system using my above solution so here's what has been working very reliably for me ---

Filename: plex-debrid.ps1 (located in plex_debrid-main directory)
$Title = "plex_debrid"
$host.UI.RawUI.WindowTitle = $Title
cd E:\Desktop\plex_debrid-main
python ./main.py

Filename: Autostart-debrid.bat (located in plex_debrid-main directory, and shortcut in Windows Startup folder)
powershell.exe -executionpolicy remotesigned -File E:\Desktop\plex_debrid-main\plex-debrid.ps1

Filename: rclone.ps1 (located in rclone directory)
$Title = "rclone"
$host.UI.RawUI.WindowTitle = $Title
cd E:\Desktop\rclone
./rclone.exe cmount realdebrid: X: --dir-cache-time 10s --no-console

Filename: Autostart-rclone.bat (located in rclone directory, and shortcut in Windows Startup folder)
powershell.exe -executionpolicy remotesigned -File E:\Desktop\rclone\rclone.ps1

Filename: Autostart-flaresolverr.bat (located in flaresolverr directory, and shortcut in Windows Startup folder)
start "" "E:\Desktop\flaresolverr\flaresolverr.exe"

**Reply by omerf111** [reply]
thank you for the write-up. but I have one problem. after executing Autostart-debrid.bat I am greeted with the main plex-debrid menu with 4 options: run, settings, ignored media, and scraper. with that plex-debrid won't work for new requests. how did you overcome this?

Found the solution: I changed the value "Show Menu on Startup" from true to false in the settings.json file in the plex_debrid-main folder.

**Reply by AndyZaa** [reply]
Good to hear! By the way, if you need to go back to the Settings menu - I can never get entering "exit" to work for me. If it's the same for you, you can edit that same settings.json "Show Menu on Startup" value back to "true" and restart plex_debrid.

**Comment by AndyZaa** [comment]
Hey @omerf111,
Yep, I had to discover that issue too - good point. In the instructions in [Setup plex_debrid Standard Version](https://github.com/itsToggle/plex_debrid#3-page_facing_up-setup-plex_debrid) in Step 3-11:
"If you dont want the main menu to show when starting the script (for an auto-run setup), navigate to "/Settings/UI Settings/show menu on startup" and set the value to "false"."
and a follow-up on:
https://github.com/itsToggle/plex_debrid/discussions/416

Enjoy - plex_debrid is awesome!
